plugins/org.jkiss.dbeaver.data.transfer/src/org/jkiss/dbeaver/tools/transfer/DataTransferSettings.java

```diff
@@ -179,7 +180,17 @@ private void initializePipes(
             // Both producers and consumers specified
             // Processor belongs to non-database nodes anyway
             if (initProducers.length != initConsumers.length) {
-                throw new IllegalArgumentException("Producers number must match consumers number");
+                // Something went wrong
+                if (initProducers.length < initConsumers.length && initProducers[0] instanceof DatabaseTransferProducer) {
+                    // In this case, we had data transfer from table(s) to a file or another container.
+                    // But, probably, table(s) were deleted, and we lost our producer(s).
+                    // Usually, consumers do not have any special info, so we can delete extra items.
+                    IDataTransferConsumer<?,?>[] initConsumersCopy = new IDataTransferConsumer<?,?>[initProducers.length];
+                    System.arraycopy(initConsumers, 0, initConsumersCopy, 0, initProducers.length);
+                    initConsumers = initConsumersCopy;
```

Comment:
You can use the `Arrays#copyOf` here.

```suggestion
                    initConsumers = Arrays.copyOf(initConsumers, initProducers.length);
```